pallets/manta-pay/src/types.rs

```diff
@@ -59,6 +56,72 @@ where
     T::decode(&mut bytes.as_slice())
 }
 
+pub const FP_ENCODE: &str = "Fp encode should not failed.";
+pub const FP_DECODE: &str = "Fp decode should not failed.";
+pub const GROUP_ENCODE: &str = "Group encode should not failed.";
+pub const GROUP_DECODE: &str = "Group decode should not failed.";
+pub const PROOF_ENCODE: &str = "Proof encode should not failed.";
+pub const PROOF_DECODE: &str = "Proof decode should not failed.";
+
+/// Field encode to byte array
+pub fn fp_encode<T>(fp: Fp<T>) -> Result<[u8; 32], scale_codec::Error>
+where
+    T: manta_crypto::arkworks::ff::Field,
+{
+    use manta_util::codec::Encode;
+    fp.to_vec()
+        .try_into()
+        .map_err(|_e| scale_codec::Error::from(FP_ENCODE))
+}
+
+/// Field decode from byte array
```

Comment:
can you add an explanation why we are changing these in the description of the PR, please

Reply:
Yeah